packages/maskbook/src/components/InjectedComponents/ToolboxHint.tsx

```diff
@@ -192,7 +194,7 @@ export function ToolboxHint(props: ToolboxHintProps) {
                 <Image src={ToolIconURLs.encryptedmsg.image} width={19} height={19} />
                 <Typography className={classes.text}>{ToolIconURLs.encryptedmsg.text}</Typography>
             </MenuItem>,
-            chainIdValid ? (
+            RedPacket_Base.enableRequirement.web3?.compositionEntryRequiredChains?.includes(chainId) ? (
```

Comment:
It's a bit verbose to add this condition. Does it make sense to make a hook for this?